### PR TITLE
mgr: fix ceph native option value types

### PR DESCRIPTION
**Pull request by liewegas**
Return Py types, just like we do with module options.

### CODE REVIEW

File: src/mgr/BaseMgrModule.cc [hunk 1]
```diff
@@ -393,11 +394,13 @@ ceph_option_get(BaseMgrModule *self, PyObject *args)
     return nullptr;
   }
 
-  std::string value;
-  int r = g_conf().get_val(string(what), &value);
-  if (r >= 0) {
+  const Option *opt = g_conf().find_option(string(what));
```
Comment (tchaikov) [inline src/mgr/BaseMgrModule.cc]: now that we already have `opt`, why not just query it for its value?
```c++
if (auto opt = g_conf().find_option(what); opt != nullptr) {
  auto value = opt->to_str();
  // ...
  return get_python_typed_option_value(opt->type, value);
} else {
  // ...
  Py_RETURN_NONE;
}
```

File: src/mgr/PyUtil.cc [hunk 2]
```diff
@@ -0,0 +1,41 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "PyUtil.h"
+#include "PythonCompat.h"
+
+PyObject *get_python_typed_option_value(
+  Option::type_t type,
+  const std::string& name,
```
Comment (tchaikov) [inline src/mgr/PyUtil.cc]: can we drop this parameter? as it's not used.

**Comment by tchaikov** [comment]
could you remove the merge commit. or just rebase the change to master?

File: src/mgr/BaseMgrModule.cc (hunk 1 above)
Comment (liewegas) [reply src/mgr/BaseMgrModule.cc]: opt->value and daemon_value are defaults; the actual values are stored in a separate map in mdconfig_t.

**Comment by liewegas** [comment]
rebased, and dropped the unused name arg

**Comment by liewegas** [comment]
retest this please

**Pull request merged**